toolbox/library/CMakeLists.txt

```diff
@@ -2,182 +2,88 @@
-        HEADERS include/WBToolbox/Block/MinimumJerkTrajectoryGenerator.h)
-
-    configure_block(BLOCK_NAME "Discrete Filter"
-        GROUP "Utilities"
-        LIST_PREFIX WBT
-        SOURCES src/DiscreteFilter.cpp
-        HEADERS include/WBToolbox/Block/DiscreteFilter.h)
-
-    # if (${ICUB_USE_IPOPT})
-    #     find_package(iDynTree REQUIRED)
-    #     add_definitions(-DWBT_USES_IPOPT)
-    #     configure_block(BLOCK_NAME "Inverse Kinematics"
-    #         GROUP "Model"
-    #         LIST_PREFIX WBT
-    #         SOURCES src/InverseKinematics.cpp
-    #         HEADERS include/InverseKinematics.h)
-    #
-    #     include_directories(SYSTEM ${iKin_INCLUDE_DIRS})
-    # endif()
+find_package(ICUB)
+if(NOT WBT_USES_ICUB)
+    option(WBT_USES_ICUB "Build blocks that depend on iCub library" ${ICUB_FOUND})
 endif()
 
-# Search qpOASES and enable block which depends on it
 find_package(qpOASES)
 if(NOT WBT_USES_QPOASES)
-    option(WBT_USES_QPOASES "Compile also qpOASES-related components" ${qpOASES_FOUND})
+    option(WBT_USES_QPOASES "Build blocks that depend on qpOASES library" ${qpOASES_FOUND})
 endif()
 
-configure_block(BLOCK_NAME "Yarp Read"
-    GROUP "Utilities"
-    LIST_PREFIX WBT
-    SOURCES src/YarpRead.cpp
-    HEADERS include/WBToolbox/Block/YarpRead.h)
-
-configure_block(BLOCK_NAME "Yarp Write"
-    GROUP "Utilities"
-    LIST_PREFIX WBT
-    SOURCES src/YarpWrite.cpp
-    HEADERS include/WBToolbox/Block/YarpWrite.h)
-
-configure_block(BLOCK_NAME "Real Time Synchronizer"
-    GROUP "Utilities"
-    LIST_PREFIX WBT
-    SOURCES src/RealTimeSynchronizer.cpp
-    HEADERS include/WBToolbox/Block/RealTimeSynchronizer.h)
-
-configure_block(BLOCK_NAME "Yarp Clock"
-    GROUP "Utilities"
-    LIST_PREFIX WBT
-    SOURCES src/YarpClock.cpp
-    HEADERS include/WBToolbox/Block/YarpClock.h)
+# ===================
+# BLOCKFACTORY PLUGIN
+# ===================
+
+# YARP blocks
+list(APPEND WBT_BLOCKS "YarpRead")
+list(APPEND WBT_BLOCKS "YarpWrite")
+list(APPEND WBT_BLOCKS "YarpClock")
+list(APPEND WBT_BLOCKS "GetLimits")
+list(APPEND WBT_BLOCKS "SetReferences")
+list(APPEND WBT_BLOCKS "GetMeasurement")
+list(APPEND WBT_BLOCKS "SetMotorParameters")
+list(APPEND WBT_BLOCKS "SimulatorSynchronizer")
+
+# iDynTree blocks
+list(APPEND WBT_BLOCKS "MassMatrix")
+list(APPEND WBT_BLOCKS "InverseDynamics")
+list(APPEND WBT_BLOCKS "DotJNu")
+list(APPEND WBT_BLOCKS "Jacobian")
+list(APPEND WBT_BLOCKS "ForwardKinematics")
+list(APPEND WBT_BLOCKS "RelativeTransform")
+list(APPEND WBT_BLOCKS "CentroidalMomentum")
+
+# Other
+list(APPEND WBT_BLOCKS "RealTimeSynchronizer")
+
+if(${ICUB_FOUND})
+    list(APPEND WBT_BLOCKS "DiscreteFilter")
+    list(APPEND WBT_BLOCKS "MinimumJerkTrajectoryGenerator")
+endif()
 
-if(WBT_USES_QPOASES)
-    configure_block(BLOCK_NAME "qpOASES"
-        GROUP "Utilities"
-        LIST_PREFIX WBT
-        SOURCES src/QpOases.cpp
-        HEADERS include/WBToolbox/Block/QpOases.h)
+if(${qpOASES_FOUND})
```

Comment:
With this change, qpOASES blocks are compiled even if `WBT_USES_QPOASES` is not enabled. I am afraid this may be confusing.

Reply:
If you want this to be the behaviour (not a big fan of it to be honest), problably it would make sense to just remove those variables then.

Reply:
Mmh I didn't realized I used the `_FOUND` here, my mistake. If you see in the bottom of this file, the `WBT_USES_` variables are used again. Thanks!